### PR TITLE
Pr/1

### PR DESCRIPTION
**Pull request by Tor101**

### CODE REVIEW

File: AutonomousAI/chat.py [hunk 1]
```diff
@@ -32,9 +32,11 @@ def chat_with_ai(prompt, user_input, full_message_history, permanent_memory, tok
     str: The AI's response.
     """
     current_context = [create_chat_message("system", prompt), create_chat_message("system", f"Permanent memory: {permanent_memory}")]
+    # print(current_context)
     current_context.extend(full_message_history[-(token_limit - len(prompt) - len(permanent_memory) - 10):])
+    # print(current_context)
     current_context.extend([create_chat_message("user", user_input)])
-
+    # print(current_context)
```
Comment (ksylvan) [inline AutonomousAI/chat.py]: Debugging line here, left in the PR?

File: AutonomousAI/chat.py [hunk 2]
```diff
@@ -46,9 +48,16 @@ def chat_with_ai(prompt, user_input, full_message_history, permanent_memory, tok
         print("----------- END OF CONTEXT ----------------")
 
     response = openai.ChatCompletion.create(
-        model="gpt-4",
+        model="gpt-3.5-turbo",
```
Comment (ksylvan) [inline AutonomousAI/chat.py]: Should this be based on if the -gpt3only flag is set? (i.e. set the model to "gpt-4" or "gpt-3.5-turbo" based on that user flag)?

File: AutonomousAI/data/prompt.txt [hunk 3]
```diff
@@ -22,7 +22,7 @@ RESOURCES:
 
 1. Internet access for searches and information gathering
 2. Long Term and Short Term memory management
-3. GPT-4 instances for text generation
+3. GPT-3.5 instances for text generation
```
Comment (ksylvan) [inline AutonomousAI/data/prompt.txt]: Same comments as before. If the user has access to GPT-4 API, they should be able to use it, yes?

**Review by ksylvan — CHANGES_REQUESTED** [review]

**Review by Tor101 — COMMENTED** [review]
Sorry I pushed while being tired.

File: AutonomousAI/data/prompt.txt (hunk 3 above)
Comment (Tor101) [reply AutonomousAI/data/prompt.txt]: Yes

File: AutonomousAI/chat.py (hunk 2 above)
Comment (Tor101) [reply AutonomousAI/chat.py]: Works both ways,

File: AutonomousAI/chat.py (hunk 1 above)
Comment (Tor101) [reply AutonomousAI/chat.py]: Sorry I pushed while being tired.

**Review by ksylvan — CHANGES_REQUESTED** [review]
Please remove all your IntelliJ files (or add them to .gitignore)

**Comment by Tor101** [comment]
I think is good to go now 💯

File: .env.template
Comment (nponeccop) [inline .env.template:1]: Why is this deleted?

File: .gitignore [hunk 4]
```diff
@@ -8,4 +8,5 @@ scripts/auto_gpt_workspace/*
 *.mpeg
 .env
 last_run_ai_settings.yaml
-outputs/*
\ No newline at end of file
+outputs/*
+.env.template
```
Comment (nponeccop) [inline .gitignore:12]: .env.template is in the repo and it isn't supposed to be modified without committing. So this doesn't look right for me

File: scripts/data/prompt.txt [hunk 5]
```diff
@@ -22,13 +22,14 @@ COMMANDS:
...
-18. Task Complete (Shutdown): "task_complete", args: "reason": "<reason>"
+18. Scientific Research: "arxiv_search", args: "input": "<search>"
+19. Task Complete (Shutdown): "task_complete", args: "reason": "<reason>"
 
 RESOURCES:
 
 1. Internet access for searches and information gathering.
 2. Long Term memory management.
-3. GPT-3.5 powered Agents for delegation of simple tasks.
+3. GPT-4 powered Agents for delegation of simple tasks.
```
Comment (nponeccop) [inline scripts/data/prompt.txt:32]: Can you explain this change?

**Review by nponeccop — CHANGES_REQUESTED** [review]

File: scripts/data/prompt.txt (hunk 5 above)
Comment (Tor101) [reply scripts/data/prompt.txt:32]: Low consumption reasons. This `agi` burns the wallet if gpt-4 is used even for simple tasks. No?

File: .gitignore (hunk 4 above)
Comment (Tor101) [reply .gitignore:12]: I listen advice of reviewer last time xd

File: .env.template
Comment (Tor101) [reply .env.template:1]: I'm new to github and idk what I'm doing some times but it said to solve my commits merge error I should remove the env template file xd and I thought if I do that it still states there when you merge. Apparently not.

**Comment by Torantulino** [comment]
This has no name or description, please fully fill these out in detail when submitting a pull request.

**Comment by Tor101** [comment]
Understood. Thank you for having patience with my pull request. You guys are awesome this project is awesome! 💯

**Comment by nponeccop** [comment]
You can change the description and reopen the PR. Contributors are welcome. Just as there are so many PRs we would like to focus on merging high-quality PRs. There are options to amend existing PR, you should try to push it through. Contribution is very rewarding.
